src/recordpage.cpp

```diff
@@ -801,9 +813,81 @@ QStringList RecordPage::applicationConfigurations()
 
 void RecordPage::appendOutput(const QString& text)
 {
+    // this regex finds ansi escapes and extracts all the numbers in it
+    QRegularExpression regex(QLatin1String("\\x1b\\[([0-9;]+)m"));
```

Comment:
`static const auto regex = QRegularExpression(QStringLiteral(...));`

Reply:
all of this code should go to a separate function instead of pushing it tightly into the widget

we would also need separate unit testing

Reply:
and instead of building lots of strings and concatenating them all the time, you should instead use a `QTextStream` and just put everything in there, then call insertHtml once on the document here in the widget

Reply:
you are also not properly escaping html-like output that may be available in the `text`